frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx

```diff
@@ -221,10 +217,20 @@ function OutputForDocModal({
     const rowsData = [];
     const docs = moveSelectedDocToTop();
     docs.forEach((item) => {
-      const output = data.find(
-        (outputValue) => outputValue?.document_manager === item?.document_id
-      );
-      const key = `${output?.prompt_id}__${output?.document_manager}__${output?.profile_manager}`;
+      const output = data.find((outputValue) => {
+        const docId =
+          outputValue?.document_manager ||
+          (outputValue?.key && getDocIdFromKey(outputValue?.key)) ||
+          null;
+        return docId === item?.document_id;
+      });
+      const key =
+        promptId +
+        "__" +
+        item?.document_id +
+        "__" +
+        (selectedProfile || profileManagerId);
+
```

Comment:
@jagadeeswaran-zipstack NIT: Can we use[ template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here? I feel like this way it will be cleaner. Please ignore if you think otherwise.